src/download/download_constructor.cc

```diff
@@ -57,28 +57,14 @@
 
 namespace torrent {
 
-struct download_constructor_is_single_path {
-  bool operator () (Object::map_type::const_reference v) const {
-    return
-      std::strncmp(v.first.c_str(), "name.", sizeof("name.") - 1) == 0 &&
-      v.second.is_string();
-  }
-};
-
-struct download_constructor_is_multi_path {
-  bool operator () (Object::map_type::const_reference v) const {
-    return
-      std::strncmp(v.first.c_str(), "path.", sizeof("path.") - 1) == 0 &&
-      v.second.is_list();
-  }
+bool
+download_constructor_is_single_path(Object::map_type::const_reference v) {
+  return v.first.find("name.") == 0 && v.second.is_string();
 };
 
-struct download_constructor_encoding_match :
-    public std::binary_function<const Path&, const char*, bool> {
-
-  bool operator () (const Path& p, const char* enc) {
-    return strcasecmp(p.encoding().c_str(), enc) == 0;
-  }
+bool
+download_constructor_is_multi_path(Object::map_type::const_reference v) {
+  return v.first.find("path.") == 0 && v.second.is_list();
```

Comment:
Using `find` is going to search the whole string, we only want to search from the beginning.

Use `rfind("path.", 0)`.